### PR TITLE
Document all the things

### PR DESCRIPTION
**Pull request by nephros**
This is just to let you know that I am in the process of generating inline documentation for all the different PM components.

It started out with me wanting to inline *comment* some code, but ended up with a almost complete qdoc  markup.

This is not finished at all (probably never will be), but it's in a previewable state, hence this draft PR.

Previews of the generated docs can be looked at here: https://htmlpreview.github.io/?https://github.com/nephros/patchmanager/blob/documentation/doc/generated/patchmanager/index.html

### CODE REVIEW

**Comment by b100dian** [comment]
This is awesome - but do you plan to merge the comments you're writing within the existing source files *and* the html output together or split them in two - comments in main, generated docs in the `documentation` branch (much like a `pages` one)?

**Comment by b100dian** [comment]
> 
> GitHub action could build the docs from here, and submit a PR or commit to the other repo.
> 

Exactly. We can define a "pages" repo just for this.
I didn't do this lately with github actions, but I did with woodpecker in codeberg, should be similar https://codeberg.org/sailmates/website/src/branch/main/.woodpecker.yml#L68 (the problem was: sources built a static website that gets commited to pages repo)

**Comment by Olf0** [comment]
> As can be seen in the footer, I am copying the Opal project in placing the generated docs under GNU FDL license. I hope this is OK for everybody.

Yes, though the GNU FDL is very old and the Creative Commons licenses are legally much phrased, plus allow for different variants to pick from ("CC BY-SA" is the equivalent of the GNU FDL).

> Good question - I have been pondering the same.

I would suggest keeping it all in one repo (this one), the chosen location `docs/generated` is fine IMHO.

Similarly I strongly suggest not to depend on multiple service providers, hence do everything on the infrastructure GitHub provides: Publish the generated documentation on GH-pages at `github.io`, for which multiple GH-action examples exist, which can be copy&pasted with a little adaption.

**Comment by b100dian** [comment]
> I would suggest keeping it all in one repo (this one), the chosen location `docs/generated` is fine IMHO.

I don't agree - keeping the same repo yes, keeping in the same branch(es) as the code is way too much compared to, say, localization files.
Same repo, "pages" branch or same org, "pages" repo.

Then again, I'm just commenting here, the doers should have precedence;)


> Similarly I strongly suggest not to depend on multiple service providers, hence do everything on the infrastructure GitHub provides: Publish the generated documentation on GH-pages at `github.io`, for which multiple GH-action examples exist, which can be copy&pasted with little adaption.

Agreed. I was just linking that a similar 'push generated stuff to different repo (or branch)' was possible on another infrastructure.
GH-pages feature is what I have in mind too, and that usually is a different repo or unrelated branch (no common ancestors).

**Comment by b100dian** [comment]
> Now I don't know if gihtub can do different "changes need four-eye review"
> settings per branch, but if not we want a seperate repo, as I'm sure no-one
> will want to review the documentation update PRs all the time.

I'm not sure documentation needs to be reviewed. If it is output from the changes in the code comments, then that can be reviewed instead.

Second topic:
Thanks for cleaning this branch - I am trying to find out what `branches: ["./doc/generated"]` and similarly `   branches: ["documentation"]` actually mean, did you already run one of the new workflows?

Third, question:
Are you aware that in this repo/org you can enable pages and specify a branch or repo for that?
I am not sure I get what "they're currently deployed on nephros.github.io" means, why not do it here?
image

File: .gitignore [hunk 1]
```diff
@@ -1,19 +1,2 @@
...
-*.moc
-patchmanager_interface.*
-patchmanager_adaptor.*
-patchmanager-dialog
-
-.build*
-src/icons/z*
-src/icons/ss
-*.list
+doc/qdoc/sailfish-qdoc-template/
```
Comment (b100dian) [inline .gitignore:1]: hmm I dont think all the other lines need to be removed here

**Review by b100dian — APPROVED** [review]
LGTM/Just do it - we will can the documentation itself later when published and patch it up

**Comment by b100dian** [comment]
I would just use the Squash and merge option in this case;)

**Pull request closed without merging**